docs/quickstart.md

```diff
@@ -3,61 +3,138 @@ hide:
+constructors that accept time-series data as inputs. These inputs can be any observed variables, such as presented 
+stimuli, head direction, position, or spike counts. 
 
-1. [How to define and fit a GLM model.](#basic-model-fitting)
-2. [What are the GLM input arguments.](#model-arguments)
-3. [How to use NeMoS with `pynapple` for pre-processing.](#pre-processing-with-pynapple)
-4. [How to use NeMoS with `scikit-learn` for pipelines and cross-validation.](#compatibility-with-scikit-learn)
+The basis objects can perform two types of transformations on the inputs:
 
-Each of these sections can be run independently of the others.
+1. **Non-linear Mapping:** This process transforms the input data through a non-linear function, 
+   allowing it to capture complex, non-linear relationships between inputs and neuronal firing rates. 
+   Importantly, this transformation preserves the properties that makes GLM easy to fit and guarantee a 
+   single optimal solution (e.g. convexity).
 
-### Basic Model Fitting
+2. **Convolution:** This applies a convolution of the input data with a bank of filters, designed to 
+   capture linear temporal effects. This transformation is particularly useful when analyzing data with 
+   inherent time dependencies or when the temporal dynamics of the input are significant.
 
-Defining and fitting a NeMoS GLM model is straightforward:
+Both transformations produce a vector of features `X` that changes over time, with a shape 
+of `(n_time_points, n_features)`.
+
+On the other hand, the `glm` module maps the feature to spike counts. It is used to learn the GLM weights, 
+evaluating the model performance, and explore its behavior on new input.
+
+### Model Fitting
+
+Here's a brief demonstration of how the `basis` and `glm` modules work together within NeMoS.
+
+#### Poisson GLM for features analysis
+
+<img src="../assets/glm_features_scheme.svg" width="100%">
+
+In this example, we'll construct a time-series of features using the basis objects, applying a non-linear mapping
+(default behavior):
+
+###### Feature Representation
 
 ```python
 import nemos as nmo
 import numpy as np
 
-# predictors, shape (n_samples, n_features)
-X = 0.2 * np.random.normal(size=(100, 1))
-# true coefficients, shape (n_features)
-coef = np.random.normal(size=(1, ))
+# define 3 input features of shape (n_samples, )
+input_1, input_2, input_3 = 0.2 * np.random.normal(size=(3, 100))
+
+# Instantiate the basis
+basis_1 = nmo.basis.MSplineBasis(n_basis_funcs=5)
+basis_2 = nmo.basis.CyclicBSplineBasis(n_basis_funcs=6)
+basis_3 = nmo.basis.MSplineBasis(n_basis_funcs=7)
+
+basis = basis_1 * basis_2 + basis_3
+
+# Generate the design matrix starting from some raw 
+# input time series, i.e. LFP phase, position, etc.
+X = basis.compute_features(input_1, input_2, input_3)
+```
+
+###### Simulate Spike Counts
+
+Let's generate some spikes from a Poisson model.
```

Comment:
should point out that most users won't do this -- they'll be getting this data from their experiment.

makes me wonder if we should show this with simulated data (vs. loading in a small example dataset)

Reply:
for a quickstart I think it is fine to use fake data. I would not use an example data here because explaining it takes more time than actually drawing some samples. 
What we should aim for is executable code,  the input is not relevant, so I would not go into the load some data route.

Reply:
I removed the section and added a simulation of the counts in the code; this should give less emphasis to this step